diff --git a/src/etc/snapshot.py b/src/etc/snapshot.py
index 82302da..a670cfa 100644
--- a/src/etc/snapshot.py
+++ b/src/etc/snapshot.py
@@ -28,7 +28,8 @@ if not src_dir:
   raise Exception("missing env var CFG_SRC_DIR")
 
 snapshotfile = os.path.join(src_dir, "src", "snapshots.txt")
-download_url_base = "https://static.rust-lang.org/stage0-snapshots"
+#download_url_base = "https://static.rust-lang.org/stage0-snapshots"
+download_url_base = "http://www.ntecs.de/downloads/rust"
 download_dir_base = "dl"
 download_unpack_base = os.path.join(download_dir_base, "unpack")
 
@@ -81,6 +82,8 @@ def get_kernel(triple):
         return "macos"
     if os_name == "freebsd":
         return "freebsd"
+    if os_name == "dragonfly":
+        return "dragonfly"
     return "linux"
 
 def get_cpu(triple):
diff --git a/src/liblibc/lib.rs b/src/liblibc/lib.rs
index bbccdf0..ee5c8ce 100644
--- a/src/liblibc/lib.rs
+++ b/src/liblibc/lib.rs
@@ -1086,6 +1086,7 @@ pub mod types {
                 pub type sighandler_t = size_t;
             }
             pub mod bsd44 {
+                use types::common::c95::{c_void};
                 use types::os::arch::c95::{c_char, c_int, c_uint};
 
                 pub type socklen_t = u32;
@@ -1158,6 +1159,17 @@ pub mod types {
                     pub sun_family: sa_family_t,
                     pub sun_path: [c_char, ..104]
                 }
+                #[repr(C)]
+                pub struct ifaddrs {
+                    pub ifa_next: *mut ifaddrs,
+                    pub ifa_name: *mut c_char,
+                    pub ifa_flags: c_uint,
+                    pub ifa_addr: *mut sockaddr,
+                    pub ifa_netmask: *mut sockaddr,
+                    pub ifa_dstaddr: *mut sockaddr,
+                    pub ifa_data: *mut c_void
+                }
+
             }
         }
 
diff --git a/src/snapshots.txt b/src/snapshots.txt
index 9fcaa30..95be8c5 100644
--- a/src/snapshots.txt
+++ b/src/snapshots.txt
@@ -1,5 +1,6 @@
 S 2014-08-29 6025926
   freebsd-x86_64 285330b798eefcc929fc94c9d0604b6172ce3309
+  dragonfly-x86_64 87b7d3d21de83dfe4c66a77fe7d3b85f14d97cf7
   linux-i386 5b57ab2dc32952dc78551a955f3c1746b2d915a3
   linux-x86_64 2624aeac3fe1b2359b61c1109e4708680e237ca5
   macos-i386 deffce32408b023bcda84f6ce338ca3de02f406b
